app/Jobs/ApplicationDeploymentJob.php

```
@@ -2230,15 +2229,15 @@ private function generate_nixpacks_env_variables()
         // Add COOLIFY_* environment variables to Nixpacks build context
-        $coolify_envs = $this->generate_coolify_env_variables();
+        $coolify_envs = $this->generate_coolify_env_variables(forBuildTime: true);
         $coolify_envs->each(function ($value, $key) {
             $this->env_nixpacks_args->push("--env {$key}={$value}");
         });
 
         $this->env_nixpacks_args = $this->env_nixpacks_args->implode(' ');
     }
 
-    private function generate_coolify_env_variables(): Collection
+    private function generate_coolify_env_variables(bool $forBuildTime = false): Collection
```

PHPMD (2.15.0)

The method generate_coolify_env_variables has a boolean flag argument $forBuildTime, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)